spectrum-network/src/network_controller.rs

```diff
@@ -103,56 +140,82 @@ impl<TPeers, TPeerManager, THandler> NetworkController<TPeers, TPeerManager, THa
 where
     THandler: Clone,
 {
-    fn init_handler(&self, role: PeerRole) -> PartialPeerConnHandler {
+    pub fn new(
+        conn_handler_conf: PeerConnHandlerConf,
+        supported_protocols: HashMap<ProtocolId, (ProtocolConfig, THandler)>,
+        peers: TPeers,
+        peer_manager: TPeerManager,
+        requests_recv: UnboundedReceiver<NetworkControllerIn>,
+    ) -> Self {
+        Self {
+            conn_handler_conf,
+            supported_protocols,
+            peers,
+            peer_manager,
+            enabled_peers: HashMap::new(),
+            requests_recv,
+            pending_actions: VecDeque::new(),
+        }
+    }
+
+    fn init_handler(&self) -> PartialPeerConnHandler {
         PartialPeerConnHandler::new(
-            role,
             self.conn_handler_conf.clone(),
             self.supported_protocols
-                .values()
-                .cloned()
-                .map(|(conf, _)| conf)
-                .collect(),
+                .iter()
+                .clone()
+                .map(|(prot_id, (conf, _))| (*prot_id, conf.clone()))
+                .collect::<Vec<_>>(),
         )
     }
 }
 
-pub fn make<TPeers, TPeerManager, THandler>(
-    conn_handler_conf: PeerConnHandlerConf,
-    supported_protocols: HashMap<ProtocolId, (ProtocolConfig, THandler)>,
-    peers: TPeers,
-    peer_manager: TPeerManager,
-) -> (
-    NetworkController<TPeers, TPeerManager, THandler>,
-    NetworkMailbox,
-) {
-    let (requests_snd, requests_recv) = mpsc::unbounded::<NetworkControllerIn>();
-    let network_controller = NetworkController {
-        conn_handler_conf,
-        supported_protocols,
-        peers,
-        peer_manager,
-        enabled_peers: HashMap::new(),
-        requests_recv,
-        pending_actions: VecDeque::new(),
-    };
-    let network_api = NetworkMailbox {
-        mailbox_snd: requests_snd,
-    };
-    (network_controller, network_api)
+impl<TPeers, TPeerManager, THandler> NetworkEvents for NetworkController<TPeers, TPeerManager, THandler> {
+    fn peer_connected(&mut self, peer_id: PeerId) {
+        self.pending_actions
+            .push_back(NetworkBehaviourAction::GenerateEvent(
```

Comment:
Who is responsible for handling this events? Also peer_disconnected, protocol_enabled, protocol_enabled

Reply:
These events can be handy in tests for NC. They provide a way to see the result of particular actions with swarm from outside.

Reply:
Not wired and not used currently tho, so we can remove them from this PR and introduce later.

Reply:
remove please